sc-memory/sc-memory/sc_action.cpp

```diff
@@ -0,0 +1,279 @@
+{
+  if (!IsInitiated())
+    SC_THROW_EXCEPTION(
+        utils::ExceptionInvalidState,
+        "Not able to get result of action " << GetActionPrettyString() << GetActionClassPrettyString()
+                                            << "` because it had not been initiated yet.");
+
+  if (!IsFinished())
+    SC_THROW_EXCEPTION(
+        utils::ExceptionInvalidState,
+        "Not able to get result of action " << GetActionPrettyString() << GetActionClassPrettyString()
+                                            << " because it had not been finished yet.");
+
+  ScIterator5Ptr const & it5 = m_context->Iterator5(
+      *this, ScType::EdgeDCommonConst, ScType::Unknown, ScType::EdgeAccessConstPosPerm, ScKeynodes::nrel_result);
+  if (!it5->Next())
+    SC_THROW_EXCEPTION(
+        utils::ExceptionInvalidState,
+        "Action " << GetActionPrettyString() << GetActionClassPrettyString() << " does not have result structure.");
+
+  m_resultAddr = it5->Get(2);
+  return m_context->ConvertToStructure(m_resultAddr);
+}
+
+ScAction & ScAction::SetResult(ScAddr const & structureAddr) noexcept(false)
+{
+  if (IsFinished())
+    SC_THROW_EXCEPTION(
+        utils::ExceptionInvalidState,
+        "Not able to set result for " << GetActionPrettyString() << GetActionClassPrettyString()
+                                      << " because it had already been finished.");
+
+  if (m_resultAddr == structureAddr)
+    return *this;
+
+  if (!m_context->IsElement(structureAddr))
+    SC_THROW_EXCEPTION(
+        utils::ExceptionInvalidParams,
+        "Not able to set structure for action " << GetActionPrettyString() << GetActionClassPrettyString()
+                                                << " because settable structure is not valid.");
+
+  if (m_context->IsElement(m_resultAddr))
+    m_context->EraseElement(m_resultAddr);
+
+  m_resultAddr = structureAddr;
+  return *this;
+}
+
+bool ScAction::IsInitiated() const noexcept
+{
+  return m_context->HelperCheckEdge(ScKeynodes::action_initiated, *this, ScType::EdgeAccessConstPosPerm);
+}
+
+bool ScAction::InitiateAndWait(sc_uint32 waitTime_ms) noexcept(false)
+{
+  if (IsInitiated())
+    SC_THROW_EXCEPTION(
+        utils::ExceptionInvalidState,
+        "Not able to initiate and wait action " << GetActionPrettyString() << GetActionClassPrettyString()
+                                                << " because it had already been initiated.");
+
+  if (IsFinished())
+    SC_THROW_EXCEPTION(
+        utils::ExceptionInvalidState,
+        "Not able to initiate and wait action " << GetActionPrettyString() << GetActionClassPrettyString()
+                                                << " because it had already been finished.");
+
+  auto wait = std::shared_ptr<ScWaiterActionFinished>(new ScWaiterActionFinished(*m_context, *this));
+  wait->SetOnWaitStartDelegate(
+      [this]()
+      {
+        m_context->CreateEdge(ScType::EdgeAccessConstPosPerm, ScKeynodes::action_initiated, *this);
```

Comment:
previously it was possible to pass here onEventClassAddr

Reply:
Now you don't have to do that here

Reply:
What can I use now to initiate action with not action_initiated subscription element?